### PR TITLE
compile.ml: Wrap cycle prim impls in `share_code`

### PR DESCRIPTION
**Pull request by nomeata**
in preparation for #3159

### CODE REVIEW

File: src/codegen/compile.ml [hunk 1]
```diff
@@ -3832,7 +3834,91 @@ module IC = struct
 
 end (* IC *)
 
+module Cycles = struct
```
Comment (ggreif) [inline src/codegen/compile.ml:3837]: This module just moved, right?

File: src/codegen/compile.ml [hunk 2]
```diff
@@ -8302,62 +8348,18 @@ and compile_prim_invocation (env : E.t) ae p es at =
...
-      get_dst ^^
-      Cycles.load_cycles env
-    )
+    SR.Vanilla, Cycles.refunded env
+
   | SetCertifiedData, [e1] ->
-    SR.unit,
-    compile_exp_vanilla env ae e1 ^^
-    IC.set_certified_data env
+    SR.unit, compile_exp_vanilla env ae e1 ^^ IC.set_certified_data env
```
Comment (ggreif) [inline src/codegen/compile.ml:8362]: was this intentional?

Comment (nomeata) [reply src/codegen/compile.ml:8362]: Yes, a bit more consistent with the lines above. Irrelevant cosmetics, though

**Review by ggreif — APPROVED** [review]
LGTM, but this is a rather sensitive area, so another pair of eyeballs would be preferable. @crusso?

**Review by crusso — APPROVED** [review]
LGTM

**Pull request merged**